layouts/DocsPage/Versions.tsx

```diff
@@ -16,18 +18,42 @@ const Versions = ({
 }: VersionsInfo) => {
   const router = useRouter();
   const [currentItem, setCurrentItem] = useState<string>(current);
-
+  const [articleList, setArticleList] = useState<LinkWithRedirectList>({});
+  const currentPage = getCurrentPageWithScope(router.asPath);
   const versions = useMemo(() => [...available].reverse(), [available]);
 
+  useEffect(() => {
```

Comment:
Q: WDYT about moving all this logic to custom hook and adding caching to it? I think right now it will be reloaded on each page navigation.

We can also make function `findExistingPage` and exportable part of this hook and reduce number of params we need to sent to in significantly this way.